addons/l10n_in_edi/models/account_edi_format.py

```diff
@@ -345,6 +360,94 @@ def _get_l10n_in_edi_line_details(self, index, line, line_tax_details, sign):
                 (line.balance + line_tax_details.get("tax_amount", 0.00)) * sign),
         }
 
+    def _l10n_in_edi_generate_invoice_json_managing_negative_lines(self, invoice, json_payload):
+        """Set negative lines against positive lines as discount with same HSN code and tax rate
+
+            With negative lines
+
+            product name | hsn code | unit price | qty | discount | total
+            =============================================================
+            product A    | 123456   | 1000       | 1   | 100      |  900
+            product B    | 123456   | 1500       | 2   | 0        | 3000
+            Discount     | 123456   | -300       | 1   | 0        | -300
+
+            Converted to without negative lines
+
+            product name | hsn code | unit price | qty | discount | total
+            =============================================================
+            product A    | 123456   | 1000       | 1   | 100      |  900
+            product B    | 123456   | 1500       | 2   | 300      | 2700
+
+        """
```

Comment:
add in the docstring that totally discounted lines are kept as 0, though

Reply:
okay